### PR TITLE
replaced regedit with winreg

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/msdeployutility.ts [hunk 1]
```diff
@@ -139,32 +138,41 @@ export async function getMSDeployFullPath() {
...
+      key:  registryKey
     })
-    .on('error', (error) => {
-        defer.reject(tl.loc("UnabletofindthelocationofMSDeployfromregistryonmachineError", error));
+
+    regKey.keys(function(err, subRegKeys) {
+        if(err) {
+            defer.reject(tl.loc("UnabletofindthelocationofMSDeployfromregistryonmachineError", err));
+        }
+        subRegKeys.sort();
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/msdeployutility.ts]: Is this sort safe to sort version number? This sort will be string sort while we need version numbers to be sorted right?

File: Tasks/AzureRmWebAppDeployment/task.json [hunk 2]
```diff
@@ -148,7 +148,7 @@
             "name": "TakeAppOfflineFlag",
             "type": "boolean",
             "label": "Take App Offline",
-            "defaultValue": "true",
+            "defaultValue": "false",
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/task.json]: Patch version has to be updated.

Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/task.json]: Done

File: Tasks/AzureRmWebAppDeployment/msdeployutility.ts (hunk 1 above)
Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/msdeployutility.ts]: Have changed logic.

**Comment by bryanmacfarlane** [comment]
This has been failing nix builds all week.  What is the status of this PR?

**Comment by kmkumaran** [comment]
Looks fine.

**Comment by mvvsubbu** [comment]
@Ajay-MS, @kmkumaran  I am closing this PR as went ahead with optional packages approach.

**Comment by mvvsubbu** [comment]
Taking care of this changes in a different PR

**Pull request closed without merging**